addons/project/static/src/project_sharing/chatter/chatter_patch.js

```diff
@@ -13,10 +13,18 @@ patch(Chatter.prototype, {
         this.orm = useService("orm");
         useSubEnv({
             projectSharingId: this.props.projectSharingId,
-            inFrontendPortalChatter: true,
         });
     },
 
+    _onMounted() {
+        super._onMounted();
+        if (this.env.inFrontendPrtalChatter) {
+            throw new Error(
+                "'inFrontendPortalChatter' is specific to the frontend chatter environment. Maybe you want to use env.projectSharingId instead?"
+            );
```

Comment:
Ok for now, but in a sense this is a misuse of hi-jacking  the `env.inFrontendPortalChatter`.

I suppose a small rework of `env.inX` to enforce using the actual `X` component in the parented tree should prevent this misuse in the future.

Reply:
Shouldn't it be something reviewers should be educated to check instead?

If anything, I'd rather we have a comment above the `inFrontendPortalChatter` definition to better clarify its purpose, and another comment above `projectSharingId` to explain why it is different.